docs/adr/0003-outsource-user-management.md

```diff
@@ -0,0 +1,89 @@
+# 3. Outsource User Management
+
+* Status: accepted <!-- optional -->
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1 <!-- optional -->
+* Date: 2020-12-09 <!-- optional -->
+
+Technical Story: [Skip account-service by talking to CS3 user-api](https://github.com/owncloud/ocis/pull/1020) <!-- optional -->
+
+## Context and Problem Statement
+
+To attach metadata like shares to users ownCloud relies on persistent, non-reassignable, unique identifiers for users (and files). Email and username can change when a user changes his name. But even the OIDC sub+iss combination may change when the IdP changes. While there is [an account porting protocol](https://openid.net/specs/openid-connect-account-porting-1_0.html) that describes how a relying party such as ownCloud should should behave, it still requires the RP to maintain its own user identifiers.
+
+## Decision Drivers <!-- optional -->
+
+* OCIS should be a single binary that can run out of the box without external dependencies like an LDAP server.
+* Time: we want to build a release candidate asap.
+
+## Considered Options
+
+* Accounts service wraps LDAP
+* GLauth wraps accounts service
+
+## Decision Outcome
+
+Chosen option: "Move accounts functionality to GLauth and name it accounts", by moving the existing accounts service file based persistence to glauth and use it as a drop in replacement for an LDAP server. The reverse index and web ui existing in the accounts service will move as well in order to make glauth a standalone, small scale user management with write capabilities.
+
+### Product summary
+- GLauth is a drop in user management for small scale deployments.
+- OCIS admins can either use the web ui to manage users in glauth or use existing tools in their IDM.
+- We hide the complexity by embedding OpenID Provider, an LDAP server and a user management web ui.
+
+### Resulting deployment options
+- Single binary: admin can manage users, groups and roles using the built in web ui (glauth)
+- External LDAP: OCIS admin needs to use existing tool to manage users
+- Separate OCIS and LDAP admin: OCIS admin relies on the LDAP admin to manage users
+
```

Comment:
What about the `two LDAP servers` scenario, where we use the customers LDAP for internal users and another one (GLAuth?) for guest accounts? Might even make sense to have the web ui for disabling guest users...

Reply:
GLauth already has a primary and fallback backend. The chain can easily be extended to multiple backends. login and single user properties can even be made in parallel. The only challenge is merging results when searching recipients in multiple ldap servers, but since 1000 results are not helpful for users we can limit the page size to 10 per ldap. Furthermore, multiple ldap servers indicate larger deployments where user lookup via ldap is restricted to exact matches, eg. the full email. In these cases merging becomes trivial again.

We may want to cache these queries anyway in order not flood ldap servers with lookup requests.

Reply:
We decided to make multiple LDAP servers possible through an additional service which works as a connector to multiple LDAP servers and exposes their user information via the GraphAPI

Reply:
I would summarize the two comments to "GLAuth abstracts multiple LDAP servers that might exist into one that looks like one to the consumer. A GraphAPI will be provided to interact with GLAuth to manage guest users (for that, at least one of the LDAPs must be read/write)".

Thoughts:
- The GLAuth must somehow expose if it allows write ops to create or edit (Guest-) Users or not.

Reply:
reworked the list